packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,276 @@
-import { defineComponent } from 'vue'
-import VaForm from './VaForm.demo.vue'
+import { userEvent } from '../../../.storybook/interaction-utils/userEvent'
+import { addText } from '../../../.storybook/interaction-utils/addText'
+import { expect } from '@storybook/jest'
+import { VaForm } from './'
+import { VaCheckbox } from '../va-checkbox'
+import { VaButton } from '../va-button'
+import { VaInput } from '../va-input'
+import { within } from '@storybook/testing-library'
 
 export default {
   title: 'VaForm',
   component: VaForm,
+  tags: ['autodocs'],
 }
 
-export const Default = defineComponent({
-  components: { VaForm },
-  template: '<VaForm/>',
+export const Default = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form ref="form">
+      <va-input :rules="[false]"/>
+    </va-form>
+    <va-button @click.prevent="$refs.form.validate()">
+      Submit
+    </va-button>
+  `,
 })
+
+export const Autofocus = () => ({
+  components: { VaForm, VaInput },
+  template: `
+    <va-form autofocus>
+      <va-input />
+    </va-form>
+  `,
+})
+
+addText(
+  Autofocus,
+  'Autofocus is broken',
+  'stale',
+)
+
+Autofocus.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const input = canvas.getByRole('textbox', { name: '' }) as HTMLElement
+
+  await step('Focus first input', async () => {
+    expect(input).toHaveFocus()
+  })
+}
+
+export const Stateful = () => ({
+  components: { VaForm, VaCheckbox },
+  template: `
+    [true]
+    <va-form stateful>
+      <va-checkbox />
+    </va-form>
+    [false]
+    <va-form>
+      <va-checkbox />
+    </va-form>
+  `,
+})
+
+Stateful.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const [firstCheckboxInput, secondCheckboxInput] = canvas.getAllByRole('checkbox', { name: '' }) as HTMLElement[]
+  const [firstCheckbox, secondCheckbox] = document.querySelectorAll('.va-checkbox__input-container')
+
+  await step('Should be checked', async () => {
+    await userEvent.click(firstCheckbox)
+    expect(firstCheckboxInput).toBeChecked()
+  })
+
+  await step('Should not be checked', async () => {
+    await userEvent.click(secondCheckbox)
+    expect(secondCheckboxInput).not.toBeChecked()
+  })
+}
+
+export const Immediate = () => ({
+  components: { VaForm, VaInput },
+  template: `
+    [true]
+    <va-form immediate>
+      <va-input :rules="[false]"/>
+    </va-form>
```

Comment:
Passing false directly prevents any interaction within the form in stories. I'd like to be able to play with the story and make sure there is no error if I typed something. Also, having a small message makes it easier to understand the story without digging into source code.

```suggestion
    <va-form immediate>
      Input is required and must show error if empty
      <va-input :rules="[(v) => v?.length > 0 || 'required']"/>
    </va-form>
```

Reply:
I meant to do this to Default demo, not immediate.